ompi/mpi/c/bindings.h

```diff
@@ -104,6 +104,18 @@ BEGIN_C_DECLS
         }                                                               \
     } while (0)
 
+
+/* check for integer overflow - needed while parts of Open MPI have not been embiggened */
+#define OMPI_CHECK_MPI_COUNT_INT_CONVERSION_OVERFLOW(RC, x) \
```

Comment:
After reading this PR and circling back to this macro -- is the purpose of this macro to cover until #13152 and #13153 are fixed?

I.e., are you imaging shipping 6.0 before these 2 issues are fixed?  If so, I'm a little curious as to why we want to merge a PR that creates technical debt.

Do we know how much work it will take to fix these 2 issues?

Reply:
Yes it is to trap MPI_Count args that are too big for our unembiggened datatype infrastructure.  I think #13152 is not too difficult since internally it is already using size_t for the relevant quantities.  We decided I thought to defer #13153 till a later release.

Reply:
Did we?  I.e., did we really decide to ship support for bigcount without really supporting bigcount?

I know we chose to do a "put in stubs for MPI_T events but just have them return 'not supported'" kind of approach, but that's for a feature that is both not widely being used, and there's a PR that has been revived to actually add the support.

Bigcount is something that apps are actively using, right?  I.e., if we ship support for the API but you can't actually use big counts, won't we just get complaints?

Reply:
For a lot of the functionality we should be okay with mpi_count args that are greater than INT_MAX.  But for datatypes that won't be the case.  There are other things that don't support "bigcount".  For instance, several OFI providers have a max message size that is on the order of INT_MAX or smaller, so for those OFI providers, we can't really support big count using the CM PML.

Reply:
There was a good deal of discussion on this PR - https://github.com/open-mpi/ompi/pull/12351 that may be relevant here.